src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -509,15 +508,18 @@ namespace cryptonote
     }
     //std::cout << "!"<< tx.vin.size() << std::endl;
 
+    bad_semantics_txes_lock.lock();
```

Comment:
Maybe. The lock's only taken for a tiny amount of time though, two set lookups. Looks like too much change right now just to potentially gain epsilon. The log's out of the lock scope btw, but good point about scope locks in general.